### PR TITLE
Add in memory support for OTEL TLS settings for certificates

### PR DESCRIPTION
**Pull request by erikbaranowski**
#### PR Description

Add in memory support for OTEL TLS settings for certificates leveraging this [PR](https://github.com/grafana/opentelemetry-collector/pull/2). An upstream issue has been opened [here](https://github.com/open-telemetry/opentelemetry-collector/issues/7538).

#### Notes to the Reviewer

were using shared otel tls docs
- otelcol.auth.oauth2
- otelcol.exporter.jaeger
- otelcol.exporter.otlp
- otelcol.exporter.otlphttp
- otelcol.extension.jaeger_remote_sampling
- otelcol.receiver.kafka

updated to use shared otel tls docs
- otelcol.receiver.jaeger
- otelcol.receiver.zipkin
- otelcol.receiver.opencensus

#### Researched type usage

TLSServerArguments
- HTTPServerArguments
  - otelcol.extension.jaeger_remote_sampling
  - *otelcol.receiver.jaeger
  - otelcol.exporter.otlp
  - *otelcol.receiver.zipkin
		
- GRPCServerArguments
  - otelcol.extension.jaeger_remote_sampling
  - *otelcol.receiver.jaeger
  - *otelcol.receiver.opencensus
  - otelcol.exporter.otlp

TLSClientArguments
- otelcol.auth.oauth2
- otelcol.receiver.kafka
	
- HTTPClientArguments
  - otelcol.exporter.otlphttp
		
- GRPCClientArguments
  - otelcol.exporter.jaeger
  - otelcol.exporter.otlp
  - otelcol.extension.jaeger_remote_sampling
  - *otelcol.receiver.jaeger

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -87,6 +87,26 @@ Main (unreleased)
 
 - Agent Management: Introduces backpressure mechanism for remote config fetching (obeys 429 request
   `Retry-After` header). (@spartan0x117)
+
```
Comment (rfratto) [inline CHANGELOG.md:90]: Sorry, I meant to merge my other PR before you opened this, can you rebase and combine the lists?

Comment (erikbaranowski) [reply CHANGELOG.md:90]: no problem I anticipated the merge would be necessary

Comment (erikbaranowski) [reply CHANGELOG.md:90]: hang tight, going to merge them git was too smart auto merging

Comment (erikbaranowski) [reply CHANGELOG.md:90]: much better

**Pull request merged**